frontend/src/metabase/dashboard/components/DashboardLeaveConfirmationModal/DashboardLeaveConfirmationModal.tsx

```diff
@@ -1,64 +1,85 @@
 import { t } from "ttag";
 
-import ConfirmContent from "metabase/components/ConfirmContent";
-import {
-  LeaveConfirmationModal,
-  LeaveConfirmationModalContent,
-} from "metabase/components/LeaveConfirmationModal";
 import { updateDashboardAndCards } from "metabase/dashboard/actions/save";
+import { useConfirmLeaveModal } from "metabase/hooks/use-confirm-leave-modal";
 import { useDispatch } from "metabase/lib/redux";
 import { dismissAllUndo } from "metabase/redux/undo";
+import { Button, Flex, Modal, Text } from "metabase/ui";
 
 import { isNavigatingToCreateADashboardQuestion } from "./utils";
 
-interface DashboardLeaveConfirmationModalProps {
+interface DashboardLeaveConfirmationModalProps extends WithRouterProps {
   isEditing: boolean;
   isDirty: boolean;
   route: Route;
 }
 
-export const DashboardLeaveConfirmationModal = ({
-  isEditing,
-  isDirty,
-  route,
-}: DashboardLeaveConfirmationModalProps) => {
-  const dispatch = useDispatch();
+export const DashboardLeaveConfirmationModal = withRouter(
+  ({
+    isEditing,
+    isDirty,
+    router,
+    route,
+  }: DashboardLeaveConfirmationModalProps) => {
+    const dispatch = useDispatch();
 
-  // TODO: should this come from props?
-  const onSave = async () => {
-    dispatch(dismissAllUndo());
-    await dispatch(updateDashboardAndCards());
-  };
+    const { opened, close, confirm, nextLocation } = useConfirmLeaveModal({
+      isEnabled: isEditing && isDirty,
+      route,
+      router,
+    });
 
-  return (
-    <LeaveConfirmationModal isEnabled={isEditing && isDirty} route={route}>
-      {({ nextLocation, onAction, onClose }) => {
-        if (isNavigatingToCreateADashboardQuestion(nextLocation)) {
-          return (
-            <ConfirmContent
-              cancelButtonText={t`Cancel`}
-              confirmButtonText={t`Save changes and go`}
-              confirmButtonPrimary
-              data-testid="leave-for-new-dashboard-question-confirmation"
-              message={t`That’ll keep things tidy.`}
-              title={t`Let’s save your changes and create your new question`}
-              onAction={async () => {
-                await onSave();
-                onAction?.();
-              }}
-              onClose={onClose}
-            />
-          );
+    const onSave = async () => {
+      dispatch(dismissAllUndo());
+      await dispatch(updateDashboardAndCards());
+      confirm?.();
+    };
+
+    const content = isNavigatingToCreateADashboardQuestion(nextLocation)
+      ? {
+          title: t`Save your changes?`,
+          message: t`You’ll need to save your changes before leaving to create a new question.`,
+          actionBtn: {
+            color: "primary",
+            message: t`Save changes`,
+          },
         }
+      : {
+          title: t`Discard your changes?`,
+          message: t`Your changes haven't been saved, so you'll lose them if you navigate away.`,
```

Comment:
Fwiw I don't think we're consistent about smart v normal apostrophes in messages like this and I wonder if that will cause a little trouble